### PR TITLE
Refactor shop logic, add islands to shop

### LINKED ISSUE #97: Add Islands To Shop / Refactor Shop Logic
## Problem
As a user I would like to be able to buy islands from the shop.

## Definition of Done
- Shop screen has an island carousel with the "Islands" label
- User can purchase islands from the shop
  - Purchased islands are stored on the server and the local db

## Subtasks
- [x] Update carousel label to read "Islands" instead of "Sounds"
- [x] Add 7 purchasable islands to test database
- [x] Populate island carousel with islands from local database
- [x] Remove logic saving islands to local context upon item purchase
- [x] Make shop methods async where possible
- [x] Refactor error logging to catch entire exception and custom message

### PR DESCRIPTION
**Pull request by iggy808**
Fixes #97

## Problem
As a user I would like to be able to buy islands from the shop.

### Definition of Done
- Shop screen has an island carousel with the "Islands" label
- User can purchase islands from the shop
  - Purchased islands are stored on the server and the local db

## Solution
- Updated carousel label to read "Islands" instead of "Sounds"
- Added 7 purchasable islands to test database
- Populated island carousel with islands from local database
- Removed logic saving islands to local context upon item purchase
- Made shop methods async where possible
- Refactored error logging to catch entire exception and custom message
- Extracted shop logic to mediator features
- Removed unnecessary dependency injections

## Documentation
WIP

## How was this tested?
This was tested manually by giving a test user enough balance to purchase each island, and then purchasing the items and ensuring they were stored on the server and local databases.

### CODE REVIEW

File: src/FocusAPI/Controllers/SocialController.cs [hunk 1]
```diff
@@ -15,10 +15,10 @@ namespace FocusAPI.Controllers
     [Route("[controller]")]
     public class SocialController : ControllerBase
     {
-        private readonly ILogger<ShopController> _logger;
+        private readonly ILogger<SocialController> _logger;
         private IMediator _mediator;
 
-        public SocialController(ILogger<ShopController> logger, IMediator mediator)
+        public SocialController(ILogger<SocialController> logger, IMediator mediator)
```
Comment (ChrispyPeaches) [inline src/FocusAPI/Controllers/SocialController.cs:21]: Nice catch!

File: src/FocusApp.Shared/Data/FocusAppContext.cs [hunk 2]
```diff
@@ -17,6 +17,8 @@ public interface IFocusAppContext
     public DbSet<Decor> Decor { get; set; }
     public DbSet<UserDecor> UserDecor { get; set; }
     public DbSet<MindfulnessTip> MindfulnessTips { get; set; }
+    public DbSet<Island> Islands { get; set; }
+    public DbSet<UserIsland> UserIslands { get; set; }
```
Comment (ChrispyPeaches) [inline src/FocusApp.Shared/Data/FocusAppContext.cs:21]: Good catch!

**Review by ChrispyPeaches — APPROVED** [review]
Looks really good! Nice catches and props for moving the Shop logic into MediatR features!

**Review by wesleypayton — APPROVED** [review]
LGTM! 🤑

**Pull request merged**